File: forge/templates/svelte-frontend-template/template/src/lib/shared/lib/design-tokens.ts
export const DesignTokens = {
	// Spacing scale
	p4: 4,
	p8: 8,
	p12: 12,
	p16: 16,
	p20: 20,
	p24: 24,
	p32: 32,
	p48: 48,
	p64: 64,

	// Icon sizes
	iconXS: 14,
	iconSM: 16,
	iconMD: 20,
	iconLG: 24,
	iconXL: 48,
	iconHero: 80,

	// Border radii
	radiusSmall: 8,
	radiusMedium: 12,
	radiusLarge: 16,
	radiusXLarge: 20,

	// Animation durations (ms)
	durationFast: 100,
	durationNormal: 200,
	durationSlow: 300,
	durationDebounce: 400,

	// Layout dimensions
	sidebarCollapsedWidth: 72,
	sidebarExpandedWidth: 240,
	sidebarIconColumnWidth: 56,
	headerHeight: 56,
	bottomNavHeight: 64,
	minMainAreaWidth: 300,
	minChatWidth: 280,
	maxChatWidth: 480,
	chatDrawerWidth: 360,
	splitterWidth: 8,
	selectionIndicatorWidth: 3,
	selectionIndicatorHeight: 20
} as const;

File: forge/templates/flutter-frontend-template/{{project_slug}}/lib/src/theme/design_tokens.dart
abstract final class DesignTokens {
  // ── Responsive breakpoints ──
  static const double compactWidth = 600;
  static const double mediumWidth = 840;
  static const double expandedWidth = 840;

  // ── Spacing scale (8px base) ──
  static const double p4 = 4;
  static const double p8 = 8;
  static const double p12 = 12;
  static const double p16 = 16;
  static const double p20 = 20;
  static const double p24 = 24;
  static const double p32 = 32;
  static const double p48 = 48;
  static const double p64 = 64;

  // ── Border radius ──
  static const double radiusSmall = 8;
  static const double radiusMedium = 12;
  static const double radiusLarge = 16;
  static const double radiusXLarge = 20;

  // ── Icon sizes ──
  static const double iconXS = 14;
  static const double iconSM = 16;
  static const double iconMD = 20;
  static const double iconLG = 24;
  static const double iconXL = 48;
  static const double iconHero = 80;

  // ── Avatar radii ──
  static const double avatarSM = 14;
  static const double avatarMD = 16;
  static const double avatarLG = 48;

  // ── Animation durations (ms) ──
  static const int durationFast = 100;
  static const int durationNormal = 200;
  static const int durationSlow = 300;
  static const int durationDebounce = 400;

  // ── Working area header ──
  static const double workingAreaHeaderHeight = 56;

  // ── Sidebar (used by Riverpod notifier, no BuildContext) ──
  static const double sidebarCollapsedWidth = 72;
  static const double sidebarExpandedWidth = 240;
  /// Horizontal padding applied to sidebar body content (ListView + header/footer).
  static const double sidebarBodyPadding = 8;
  /// Width of the icon column inside the sidebar. All icons/avatars center
  /// within this column so they align vertically in both collapsed and expanded modes.
  /// Equals `sidebarCollapsedWidth - sidebarBodyPadding * 2`.
  static const double sidebarIconColumnWidth = sidebarCollapsedWidth - sidebarBodyPadding * 2; // 56

  // ── Legacy (kept for backward compatibility) ──
  static const double navRailWidth = 72;
  static const double navRailExtendedWidth = 256;
}
